### PR TITLE
Clean the code in `limit.rs`.

### PR DESCRIPTION
**Pull request by HaoYang670**
Signed-off-by: remzi <13716567376yh@gmail.com>

# Which issue does this PR close?

None.

# Rationale for this change

# What changes are included in this PR?

1. Remove the `current_skip` and `current_fetch` from `LimitStream` because they are redundant.
2. Remove the public function `truncate_batch` and merge its functionality into `stream_limit`. Because `truncate_batch` is    not performant when `row_num == 0` and it's only used in `stream_limit`

# Are these changes tested?
Current tests cover the changes.

# Are there any user-facing changes?
No.

### CODE REVIEW

**Comment by liukun4515** [comment]
cc @jackwener

File: datafusion/core/src/physical_plan/limit.rs [hunk 1]
```diff
@@ -365,30 +365,17 @@ impl ExecutionPlan for LocalLimitExec {
...
-        .collect();
-
-    RecordBatch::try_new(batch.schema(), limited_columns).unwrap()
-}
-
 /// A Limit stream skips `skip` rows, and then fetch up to `fetch` rows.
 struct LimitStream {
-    /// The number of rows to skip
+    /// The remaining number of rows to skip
     skip: usize,
```
Comment (jackwener) [inline datafusion/core/src/physical_plan/limit.rs:371]: The meaning of`skip` and `fetch` has changed, we should rename them.

File: datafusion/core/src/physical_plan/limit.rs [hunk 2]
```diff
@@ -420,47 +405,52 @@ impl LimitStream {
...
-                Poll::Ready(Some(Ok(batch)))
-                    if batch.num_rows() > 0 && self.current_skipped == self.skip =>
-                {
-                    break poll
+                Poll::Ready(Some(Ok(batch))) => {
+                    if batch.num_rows() > 0 && self.skip == 0 {
+                        break poll;
+                    } else {
+                        // continue to poll input stream
+                    }
```
Comment (jackwener) [inline datafusion/core/src/physical_plan/limit.rs:424]: more clear 👍

**Review by jackwener — APPROVED** [review]
make sense to me❤️.
This PR merge 
-`fetch` and `current_fetched` 
-`skip` and `current_skipped`

File: datafusion/core/src/physical_plan/limit.rs (hunk 1 above)
Comment (alamb) [reply datafusion/core/src/physical_plan/limit.rs:371]: Oh, I see this was not yet done -- @HaoYang670  can you please do this as a follow on PR? If you don't have time I will do so myself

Comment (HaoYang670) [reply datafusion/core/src/physical_plan/limit.rs:371]: Sure, I'll file a follow-up.

**Pull request merged**